resource/js/vocab-search.js

```diff
@@ -6,49 +6,246 @@ const vocabSearch = Vue.createApp({
-      const searchUrl = vocabHref + 'search?q=' + this.searchTerm + langParam
+      const searchUrlParams = new URLSearchParams({ clang: window.SKOSMOS.content_lang, q: this.searchTerm })
+      if (this.selectedLanguage === 'all') searchUrlParams.set('anylang', 'on')
+      const searchUrl = vocabHref + 'search?' + searchUrlParams.toString()
       window.location.href = searchUrl
     },
     changeLang () {
       window.SKOSMOS.content_lang = this.selectedLanguage
-      // TODO: Impelemnt partial page load to change content according to the new content language
+      // TODO: Implement (a normal) page load to change content according to the new content language
+    },
+    resetSearchTermAndHideDropdown () {
+      this.searchTerm = ''
+      this.renderedResultsList = []
+      this.hideDropdown()
+    },
+    /*
+     * Hide the autocomplete result list if the user clicks outside the list
+     */
+    onClickOutside (event) {
+      const listener = document.getElementById('headerbar-search')
+      // Check if the clicked element is outside your element
+      if (listener && !listener.contains(event.target)) {
+        this.hideDropdown()
+      }
+    },
+    /*
+     * Show the existing autocomplete list if it was hidden by onClickOutside()
+     */
+    showAutoComplete () {
+      const element = document.getElementById('search-autocomplete-results')
+      element.classList.add('show')
     }
   },
   template: `
-    <div class="d-flex mb-2 my-auto ms-auto">
-      <div class="input-group" id="search-wrapper">
+    <div class="d-flex my-auto ms-auto">
+      <div class="d-flex justify-content-end input-group ms-auto" id="search-wrapper">
         <select class="btn btn-outline-secondary dropdown-toggle" data-bs-toggle="dropdown-item" aria-expanded="false"
           v-model="selectedLanguage"
           @change="changeLang()"
         >
           <option class="dropdown-item" v-for="(value, key) in languageStrings" :value="key">{{ value }}</option>
         </select>
-        <input type="search" class="form-control" aria-label="Text input with dropdown button" placeholder="Search..."
-          v-model="searchTerm"
-          @input="autoComplete()"
-          @keyup.enter="gotoSearchPage()"
-        >
-        <button id="clear-button" class="btn btn-danger" type="clear" v-if="searchTerm" @click="searchTerm = ''">
+        <span id="headerbar-search" class="dropdown">
+          <input type="search"
+            class="form-control"
+            id="search-field"
+            aria-expanded="false"
+            autocomplete="off"
+            data-bs-toggle=""
+            aria-label="Text input with dropdown button"
+            placeholder="Search..."
+            v-model="searchTerm"
+            @input="autoComplete()"
+            @keyup.enter="gotoSearchPage()"
+            @click="showAutoComplete()">
+          <ul id="search-autocomplete-results" class="dropdown-menu"
```

Comment:
I would set the class of `search-autocomplete-results` with Vue instead of manipulating DOM directly, for example like:
```suggestion
          <ul id="search-autocomplete-results" class="dropdown-menu" :class="{ 'show': showDropdown }"
```